### PR TITLE
zettlr: 3.0.2 -> 3.2.0, respect `NIXOS_OZONE_WL`

### LINKED ISSUE #340432: Zettlr does not respect `NIXOS_OZONE_WL`
https://github.com/NixOS/nixpkgs/blob/6f6c45b5134a8ee2e465164811e451dcb5ad86e3/pkgs/applications/misc/zettlr/generic.nix

Since Zettlr is an Electron app, it needs to be run with certain command-line arguments on Wayland (specifically `--ozone-platform-hint=auto`). In NixOS, setting the `NIXOS_OZONE_WL` environment variable to `1` should automatically tell packages to use this command-line argument, but this requires integrating it into the package file (see https://github.com/NixOS/nixpkgs/blob/6f6c45b5134a8ee2e465164811e451dcb5ad86e3/pkgs/applications/networking/browsers/brave/make-brave.nix#L188) for an example. The Zettlr package does not currently do this, which results in pixelated and glitchy graphics on Wayland unless it is manually run with the correct command-line arguments. This should be a pretty simple fix, as it has been implemented before for many other Electron apps by creating a binary wrapper.

### PR DESCRIPTION
**Pull request by ryand56**
## Description of changes

Updates Zettlr to 3.2.0. Now uses NIXOS_OZONE_WL for Wayland displays.
https://github.com/Zettlr/Zettlr/releases/tag/v3.2.0
Closes #340432

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/zettlr/generic.nix [hunk 1]
```diff
@@ -19,21 +21,27 @@ in
...
 
-  extraPkgs = pkgs: [ pkgs.texliveMedium pkgs.pandoc ];
+  extraPkgs = pkgs: [
+    pkgs.texliveMedium
+    pkgs.pandoc
+  ];
 
   extraInstallCommands = ''
+    source "${makeWrapper}/nix-support/setup-hook"
+    wrapProgram $out/bin/zettlr \
```
Comment (Aleksanaa) [inline pkgs/applications/misc/zettlr/generic.nix:31]: Is it being wrapped two times?

Comment (ryand56) [reply pkgs/applications/misc/zettlr/generic.nix:31]: Ah, I'm not sure as most AppImages have the setup hook and wrapProgram.

Comment (Aleksanaa) [reply pkgs/applications/misc/zettlr/generic.nix:31]: `ls -la ./result/bin` ?

Comment (ryand56) [reply pkgs/applications/misc/zettlr/generic.nix:31]: Is this normal?
```
total 16
dr-xr-xr-x 2 root root 4096 Dec 31  1969 .
dr-xr-xr-x 4 root root 4096 Dec 31  1969 ..
-r-xr-xr-x 1 root root  279 Dec 31  1969 zettlr
lrwxrwxrwx 1 root root   62 Dec 31  1969 .zettlr-wrapped -> /nix/store/igi5gk4k3nyszmw53rsn2pfhwnxj85c4-zettlr-3.2.0-bwrap
```

**Comment by quantum9Innovation** [comment]
@Aleksanaa @tfmoraes any update on this? Would it be possible to merge this into unstable or unstable-small as a hotfix for #340432?

**Pull request merged**